### PR TITLE
Fix several tests (CTest) on Windows

### PR DESCRIPTION
**Pull request by bellenot**
- Add a couple of missing symbols into the export list
- Fix several test failing on Windows, mostly due to files staying open (creates file access issues on Windows)
- Filter-out several unsupported tests (will add them back once the underlying issues are fixed)

### CODE REVIEW

**Comment by bellenot** [comment]
> Would be good to have a marker `-DENABLE_BROKEN_WINDOWS_TESTS=On` (with accompanying `CXXFLAGS+=-DENABLE_BROKEN_WINDOWS_TESTS=1` in GNU-make speak) to tag places we should revisit. Otherwise: LGTM, congrats!

Thanks! Done. But as explained in the last commit, I had to keep some part of the code commented out, to prevent compilation errors

File: cmake/modules/SetUpWindows.cmake [hunk 1]
```diff
@@ -47,6 +47,9 @@ elseif(MSVC)
     set(CMAKE_C_FLAGS   "-nologo -I${CMAKE_SOURCE_DIR}/build/win -FIw32pragma.h -FIsehmap.h ${BLDCFLAGS} -EHsc- -W3 -D_WIN32 -DNOMINMAX")
     install(FILES ${CMAKE_SOURCE_DIR}/build/win/w32pragma.h  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} COMPONENT headers)
     install(FILES ${CMAKE_SOURCE_DIR}/build/win/sehmap.h  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} COMPONENT headers)
+    if(win_broken_tests)
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_ENABLE_BROKEN_TESTS")
```
Comment (Axel-Naumann) [inline cmake/modules/SetUpWindows.cmake]:
```suggestion
      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DR__ENABLE_BROKEN_WIN_TESTS")
```
Rationale:
- the test isn't broken, except on Windows.
- starting with `_` is evil; let's stick to usual ROOT naming convention for CPP macros and call it `R__...`

File: core/metacling/test/TClingTests.cxx [hunk 2]
```diff
@@ -191,7 +191,7 @@ static std::string MakeDepLibsPlatformIndependent(llvm::StringRef libs) {
    return llvm::StringRef(result).rtrim();
 }
 
-#ifndef _MSC_VER
+#if !defined(_MSC_VER) || defined(_ENABLE_BROKEN_TESTS)
```
Comment (Axel-Naumann) [inline core/metacling/test/TClingTests.cxx]:
```suggestion
#if !defined(_MSC_VER) || defined(R__ENABLE_BROKEN_WIN_TESTS)
```

File: math/mathmore/test/StatFunction.cxx [hunk 3]
```diff
@@ -109,6 +109,7 @@ void StatFunction::TestInverse1(RootFinder::EType algoType)
 }
 
 #ifndef _MSC_VER
```
Comment (Axel-Naumann) [inline math/mathmore/test/StatFunction.cxx]: Should this not be marked `R__ENABLE_BROKEN_WIN_TESTS` then? We need to fix the code...

Comment (bellenot) [reply math/mathmore/test/StatFunction.cxx]: As I said, the code doesn't even compile...

**Comment by bellenot** [comment]
@Axel-Naumann Everything should be OK now. Can you check and let me know? Thanks!

**Pull request merged**